### ISSUE TITLE
[batch] Support uploading local files much like we upload a script file

### ISSUE THREAD

**Issue by danking** [hail-is/hail#14354]
### What happened?

The `hailtop.batch` client uploads a script file when the command becomes too large. This functionality frees users from thinking about the size of their commands. It's a great abstraction!

The client *does not* provide a mechanism to automatically upload local files or local directories which means that users must messily combine all their code and supporting data into one big file. For example,

```python
local_in = hb.read_input('/path/to/local/script.sh')
local_data = hb.read_input('/path/to/small/local/reference.dat')

j = b.new_job()
j.command(f'bash {local_in} {local_data} {other_job.out_file} > {j.out_file}')
```

It need not necessarily be `hb.read_input`, but it does seem like a good way to re-use an extant interface. `hailtop.batch` should upload those files when it uploads large script files and then download them to the appropriate jobs.

### Version

0.2.128

**Comment by danking** [comment]
@jmarshall @illusional Please feel free to comment if I didn't fully capture the request!

**Comment by cassimons** [comment]
Thanks @danking. A feature like this would let us clean up some parts of our pipelines significantly. Our current workarounds leave a bad taste in the mouth whenever I see them.

**Comment by jmarshall** [comment]
Yes, that's the feature we were thinking of. I've got a branch mocking this up in our value-added Batch wrapper class that uses a different name:

```python
local_in = hb.store_file('local_script.sh')
[…]

j = b.new_job()
j.command(f'bash {local_in} … > {j.out_file}')
```

to emphasise that it stores a copy of the local file in a temporary bucket as well as making it available via `read_input()`, and because `hb.read_input('localfilepath')` probably already means something when running on the local backend. But further bikeshedding is warranted…

Some other wrinkles we came across:

* Users are likely to use this with a relative local path, so hopefully this denotes somewhere sensible w.r.t. the location of the Python source file it's invoked from.
* It would be good if there was some way to transmit the executable bit so you could drop `bash …` from command invocations like this one and invoke the script directly.
